plugins/net_plugin/net_plugin.cpp

```diff
@@ -3959,10 +3962,22 @@ namespace eosio {
    }
 
    // called from other threads including net threads
-   void net_plugin_impl::broadcast_vote_message(uint32_t connection_id, vote_result_t status, const vote_message_ptr& msg) {
-      fc_dlog(vote_logger, "connection - ${c} on voted signal: ${s} block #${bn} ${id}.., ${t}, key ${k}..",
-                ("c", connection_id)("s", status)("bn", block_header::num_from_id(msg->block_id))("id", msg->block_id.str().substr(8,16))
-                ("t", msg->strong ? "strong" : "weak")("k", msg->finalizer_key.to_string().substr(8, 16)));
+   void net_plugin_impl::broadcast_vote_message(uint32_t connection_id, vote_result_t status,
+                                                const vote_message_ptr& msg,
+                                                const finalizer_authority_ptr& active_auth,
+                                                const finalizer_authority_ptr& pending_auth) {
+      auto get_desc = [&]() -> std::string {
+         if (active_auth)
+            return active_auth->description;
+         if (pending_auth)
+            return pending_auth->description;
+         return "unknown";
+      };
```

Comment:
avoiding string copies
```suggestion
       auto get_desc = [&]() -> const std::string& {
         static constexpr std::string unknown {"unknown"};
         if (active_auth)
            return active_auth->description;
         if (pending_auth)
            return pending_auth->description;
         return unknown;
      };
```

Reply:
I don't think this avoids any string copies over the existing. Both will create a string in the `mutable_variant_object` `operator()( std::string key, T&& var )` `variant( std::string val )` constructor via RVO once I add `return std::string{"unknown");`

Reply:
right, my bad!